src/main/java/io/dropwizard/kafka/KafkaProducerFactory.java

```diff
@@ -74,6 +74,17 @@
     @JsonProperty
     protected int batchSize = 16384;
 
+    @Min(0)
+    @JsonProperty
+    protected int lingerMs = (int) Duration.milliseconds(0l).toMilliseconds();
```

Comment:
@PeterGroesbeck Oh, I had meant something like this:

```
@MinDuration(value = 0)
@JsonProperty
protected Duration linger = Duration.milliseconds(0L);
```
And the same for `requestTimeout`

Reply:
Oh! My mistake, thanks for clarifying. 

I'm getting some unit test failures when I try and convert these fields to `Duration`:
```
shouldBuildABasicKafkaProducer(io.dropwizard.kafka.BasicKafkaProducerFactoryTest): Invalid value 0 milliseconds for configuration linger.ms: Expected value to be a 32-bit integer, but it was a io.dropwizard.util.Duration
```

Same with requestTimeout, what am I missing? Thanks again!

Reply:
Update: Try as I might I can't seem to get the Duration classes to play nicely with Kafka's internal libraries. I went ahead and pushed an update that reverted to using `int` and `long` as well as updating the Kafka library to `2.3.0` per the other open PR: https://github.com/dropwizard/dropwizard-kafka/pull/25

This is due to the `linger.ms` type changing from `int` to `long` between `2.1.1.` and `2.3.0`.

Let me know what you think? Thank you so much for taking the time to review this.

Reply:
Let’s keep the Kafka client on 2.1.1 for now, and I’ll work on updating up separately